acator/authn/authncmd.go

```diff
@@ -67,8 +68,14 @@ func (ac *Cmd) Exec(_ io.Writer) (r Result, err error) {
 	acator.Counter = uint32(ac.Counter)
 	name = ac.UserName
 	urlStr = ac.Url
-	originURL := err2.URL.Try(url.Parse(urlStr))
-	acator.Origin = *originURL
+	if ac.Origin != "" {
+		originURL := err2.URL.Try(url.Parse(ac.Origin))
+		acator.Origin = *originURL
+	} else {
+		originURL := err2.URL.Try(url.Parse(urlStr))
+		acator.Origin = *originURL
+	}
+	origin = ac.Origin
```

Comment:
Probably it's because I am not familiar with this codebase but it seems to me we are setting now empty string to request origin if ac.Origin is not set?

Reply:
excellent catch! even at this second time, I made a mistake between `acator.Origin` and `ac.Origin`.. still weird it seemed to work in those cases when `ac.Origin` wasn't given and it should. back to the drawing board...

Reply:
hmmm... maybe it's because duolabs wont use then header value, only those authenticator messages where the origin is as well.. still weird..